src/gpuarray_collectives_cuda_nccl.c

```diff
@@ -0,0 +1,353 @@
+#include <assert.h>
+#include <stdlib.h>
+
+#include <nccl.h>
+
+#include "gpuarray/buffer_collectives.h"
+#include "gpuarray/util.h"
+#include "gpuarray/error.h"
+#include "gpuarray/config.h"
+
+#include "private.h"
+#include "private_cuda.h"
+
+#define NCCL_CHKFAIL(ctx) \
+do { \
+  if ((ctx)->nccl_err != ncclSuccess) \
+    return GA_NO_ERROR; \
+  return GA_COMM_ERROR; \
+} while (0)
+
+#define NCCL_EXIT_ON_ERROR(ctx, cmd) \
+do { \
+  (ctx)->nccl_err = (cmd); \
+  if ((ctx)->nccl_err != ncclSuccess) { \
+    cuda_exit((ctx)); \
+    return GA_COMM_ERROR; \
+  } \
+} while (0)
+
+#define GA_CHECK(cmd) \
+do { \
+  int err = (cmd); \
+  if (err != GA_NO_ERROR) \
+    return err; \
+} while (0)
+
+
+extern const gpuarray_buffer_ops cuda_ops;
+
+void comm_clear(gpucomm* comm)
```

Comment:
So I should declare/define as static whatever I choose not to expose. So pretty much every helper function in this file?